server/app/driver/driver.py

```diff
@@ -42,45 +42,12 @@ def _load_data(data):
         pass
 
     @abstractmethod
-    def filter_dataframe(self, filter):
```

Comment:
I thought it was quite specific to ScanPy implementation, as it assumes knowledge of the use of masks (which is a dataframe/numpy feature).   If other (non-Scanpy) back-ends want to use some other intermediate means of compiling filters, they shouldn't have to implement this code.   TL;DR - I'd leave it out of the driver.

Reply:
OK, I agree with that. I feel like filtering is a core feature of the engine and we might want to flag this for an offline discussion of whether some filtering belongs in the driver. But you're right about _filter_to_mask not belonging in driver.py

Reply:
It occurred to me last night that filtering was in the driver because we were filtering in rest.py, now that we are doing the serialization at the engine level there is no reason to have this in the driver.